src/pages/organize/[orgId]/campaigns/[campId]/callassignments/[callAssId]/index.tsx

```diff
@@ -54,19 +60,32 @@ const AssignmentPage: PageWithLayout<AssignmentPageProps> = ({
     return null;
   }
 
-  const statsSum =
-    model.getStats().blocked + model.getStats().done + model.getStats().ready;
+  const stats = model.getStats();
 
-  const colors = statsSum > 0 ? [ORANGE, GREEN, BLUE] : [GRAY, GRAY, GRAY];
+  const data = model.getData();
 
-  const { blocked, ready, done } = model.getStats();
+  const targetingDone = !!data.target.filter_spec?.length;
 
-  const values = statsSum > 0 ? [blocked, ready, done] : [1, 1, 1];
+  const colors = targetingDone ? [ORANGE, GREEN, BLUE] : [GRAY, GRAY, GRAY];
+  const statusBarStatsList = targetingDone
```

Comment:
What happens if targeting is done, but all values are zero? That would render a short bar with colors now, wouldn't it?

The fact that this logic is a little bit complex tells me that it should (at least partially) be moved into the model instead, so that it can be taken out of the component and easily tested.

Maybe there should be a property (getter?) on the model called `hasTargets` or something, which returns `!!data.target.filter_spec?.length && statsSum > 0`.